server/src/main/java/org/opensearch/index/mapper/TextFieldMapper.java

```diff
@@ -611,7 +618,16 @@ public ValueFetcher valueFetcher(QueryShardContext context, SearchLookup searchL
 
         void setAnalyzer(NamedAnalyzer delegate) {
             setIndexAnalyzer(
-                new NamedAnalyzer(delegate.name(), AnalyzerScope.INDEX, new PrefixWrappedAnalyzer(delegate.analyzer(), minChars, maxChars))
+                new NamedAnalyzer(
+                    delegate.name(),
+                    AnalyzerScope.INDEX,
+                    new PrefixWrappedAnalyzer(
+                        delegate.analyzer(),
+                        minChars,
+                        maxChars,
+                        parentField.indexAnalyzer().getPositionIncrementGap(parentField.name())
```

Comment:
Question please: why we use `parentField.indexAnalyzer().getPositionIncrementGap(parentField.name())` instead of passing the configured `positionIncrementGap` in private `TextFieldMapper::buildPrefixMapper`? 

```java
private PrefixFieldMapper buildPrefixMapper(BuilderContext context, FieldType fieldType, TextFieldType tft) {
   ... 
   PrefixFieldType prefixFieldType = new PrefixFieldType(tft, fullName + "._index_prefix", indexPrefixes.get(),  positionIncrementGap.get());
   ...
}
```

Reply:
That's because `positionIncrementGap` can also be defined in analyzer, if `positionIncrementGap` in mapping is not defined, then the value will be [fetched from the analyzer](https://github.com/opensearch-project/OpenSearch/blob/8673fa937db405b8d614f8d4a02c0aa52587c037/server/src/main/java/org/opensearch/index/analysis/NamedAnalyzer.java#L116), just like this:
```
PUT test11
{
  "settings": {
    "number_of_shards": 1,
    "number_of_replicas" : "0",
    "analysis" : {
          "analyzer" : {
            "custom_analyzer" : {
              "filter" : [
                "lowercase"
              ],
              "type" : "custom",
              "position_increment_gap" : "150",
              "tokenizer" : "standard"
            }
          }
        }
  },
  "mappings": {
    "properties": {
      "applications": {
        "properties": {
          "name": {
            "type": "text",
            "index_prefixes":{},
            "analyzer": "custom_analyzer"
          }
        }
      }
    }
  }
}
```
, so we can use the value of positionIncrementGap in parent field which was correctly set before: https://github.com/opensearch-project/OpenSearch/blob/8673fa937db405b8d614f8d4a02c0aa52587c037/server/src/main/java/org/opensearch/index/mapper/TextFieldMapper.java#L414

Reply:
Ah I see, thank you

Reply:
With the explanation above, I have different question now: why we don't use this index analyzer than when building prefix mapper?

```
PrefixFieldType prefixFieldType = new PrefixFieldType(tft, fullName + "._index_prefix", indexPrefixes.get());
prefixFieldType.setAnalyzer(tft.indexAnalyzer());
```

Than `setAnalyzer` would look like this:

```
        void setAnalyzer(NamedAnalyzer delegate) {
            setIndexAnalyzer(
                new NamedAnalyzer(delegate.name(), AnalyzerScope.INDEX, new PrefixWrappedAnalyzer(delegate.analyzer(), minChars, maxChars,
                        delegate.getPositionIncrementGap(name())))
            );
        }
```

Seems like logical, no?

Reply:
That makes sense, I've changed that, thanks!